high_performance/modules/high_performance/include/klepsydra/high_performance/event_loop.h

```diff
@@ -92,11 +94,11 @@ class EventLoop {
      * @brief start start consumer thread
      */
     void start() {
-        if (_isStarted) {
+        if (_isStarted.load(std::memory_order_acquire)) {
             return;
         }
+        _isStarted.store(true, std::memory_order_relaxed);
```

Comment:
This will be enough or we should use 'memory_order_seq_cst/memory_order_release' facing other threads?

Reply:
I have to fix another instance of atomic boolean variables in mem_core & high_performance files but I wanted to know if this fix works for you. I cannot replicate the error you get in the master branch, so would appreciate if you could let me know.

Reply:
similar behaviour with this. It seems that it is necessary some time between starting {server, clientServerMemProv and client} modules. Additionally, we still have 3 threads: commHandler, eventLoop and rosApp  (which launches driver, service, server, client, provider...)

Reply:
Updated to use compare_exchange_strong so that the atomic is toggled in atomically. Can you check with this?

Reply:
with yesterday, and with today's 11:56h change(latest uploaded), there are still 3 threads when launching simulation. Im reviewing our code for those instances but didnt find yet

Reply:
update: fixed issue in kpsr-bst, segfault was there, when receiving sensors_t events